### DISCUSSION TITLE
What is scope?

### DISCUSSION THREAD

**Discussion by kulak** [filebrowser/filebrowser#1323] [Q&A] [answered]
When I create a user I can assign scope with default set to ".".  It is not quite clear what it is and how to use it.

**Comment by Frank4411** [comment, marked as answer]
From my understanding Scope is like a relative path that account, like a home directory for a given user. If your scope is "." then that is the root path of your filebrowser instance that you defined with the "-r" flag.

See "filebrowser --help"
-r, --root string                        root to prepend to relative paths (default ".")

For example (in a linux instance):
If I run "filebrowser -r /home/testuser/" then my scope of "." would also be "/home/testuser" within the filebrowser web interface.

The doc page might be able to elaborate a bit more on this. I personally would recommend just adding some test files in and making a few different user accounts. I was able to sort most of that out just by playing around with the settings and adding in test files.
https://filebrowser.org/cli/filebrowser

**Comment by o1egl** [comment]
The user scope is a root directory that will be set for that user. For example, if you change the user scope to `/some/folder` user can access anything under `/some/folder` but not the top or sibling folders.

**Comment by mrmay-dev** [comment]
Noob here. I've set the root directory to my user home folder `/home/dietpi`. I get a 403 Forbidden message.

I presume this has to do with granting permissions to FileBrowser. And I'm sure it's done with CHMOD.

Would someone be so kind as to tell me how to do this?

Thanks in advance!

**Comment by dausruddin** [comment]
I am not sure if this is normal, but scope won't be applied until I re-login. Might help someone who scratching their head thinking why scope won't work.

**Comment by adeverteuil** [comment]
Is the scope `/` synonymous with `.`?

**Comment by DinoChiesa** [comment]
From my exploration, 

This does not work: 
```
/bin/filebrowser --database "$FB_DBPATH" users add "example.person@example.org" "passwordz" \
   --perm.download=true --perm.create=false --scope /data
```

Even if the /data directory exists, it fails, with 
```
Error: failed to create user home dir: [/data]: mkdir /data: file does not exist
```

But if I omit the scope from `users add` and immediately follow `users add` with `users update`, with a scope... it works. Like this: 
```
/bin/filebrowser --database "$FB_DBPATH" users add "example.person@example.org" "passwordz" \
  --perm.download=true --perm.create=false
/bin/filebrowser --database "$FB_DBPATH" users update "example.person@example.org" --scope /data
```

This is using filebrowser/filebrowser:latest in  a Dockerfile.  I did not build my own filebrowser image, so these observations might be out of date or not applicable to the current filebrowser release, if the filebrowser image is out of date.
